### PR TITLE
[KED-3064] Hero component

### PR DESCRIPTION
**Pull request by rashidakanchwala**
- This creates the hero component and all necessary elements, including the intro video.
- A low-level media component has been created as well.
- Spacing above/below the Hero component shouldn't be given much weight right now, as we'll lay things about better once we have more components for composition.
- The mobile and desktop sizes should be sound. The tablet might need some tweaking, though let's wait for design to see what we've done before changing anything, unless something is really egregious, of course.
- Note: there are a lot of changes in the `package-lock` file. I think is because Rashida ran `npm install` with an older version of NPM.

### CODE REVIEW

File: components/hero/hero.tsx
Comment (studioswong) [inline components/hero/hero.tsx:11]: You might want to set the width of this text for the mobile breakpoint as I notice that the width of this text on iphone is longer than it should on the design: 

image

image

File: components/hero/hero.module.scss
Comment (studioswong) [inline components/hero/hero.module.scss:57]: Please rename this to `get-started` according to BEM convention

File: components/hero/hero.tsx
Comment (studioswong) [inline components/hero/hero.tsx:22]: see above comment

**Review by studioswong — CHANGES_REQUESTED** [review]
Thanks for setting this up, the site looks good overall.

Please see below comments related to the style and tests. Please also add a test to test the main page ( `index.tsx`), especially to replace the tests set up in `container.test.tsx` now that we are relying on this page to display the site content rather than the container component.

File: components/hero/hero.test.tsx
Comment (tynandebold) [reply components/hero/hero.test.tsx]: I've added some tests now.

File: components/media/media.tsx
Comment (tynandebold) [reply components/media/media.tsx:12]: Good call. Added.

File: components/hero/hero.tsx
Comment (tynandebold) [reply components/hero/hero.tsx:4]: Yeah good question. It's because we're using CSS modules, which Next includes as an out-of-the-box way to [style components](https://nextjs.org/docs/basic-features/built-in-css-support#adding-component-level-css). This method of importing and then referencing the style object from the file is how you apple the class names and styles.

File: components/hero/hero.tsx
Comment (tynandebold) [reply components/hero/hero.tsx:12]: Unfortunately we can't do that with CSS modules.

File: components/media/media.tsx
Comment (tynandebold) [reply components/media/media.tsx:3]: Same thing here, too: because CSS modules.

File: components/hero/hero.tsx
Comment (tynandebold) [reply components/hero/hero.tsx:11]: Oh great call. Updated now.

File: components/hero/hero.module.scss
Comment (tynandebold) [reply components/hero/hero.module.scss:57]: I think we can avoid that here. The CSS modules convention is to use [camel case](https://github.com/css-modules/css-modules#naming), as used above. And also, since all classes are locally scoped to the file, BEM isn't too necessary in our case.

**Review by rashidakanchwala — COMMENTED** [review]
Looks great @tynandebold ! Thanks! 
I can't approve it because it's my PR! So it will have to be Susanna and you!

**Review by studioswong — COMMENTED** [review]
Thanks for adding the tests, looks good!

One thing I observed when playing around with the responsiveness of the site - the content (text, etc) really cuts to the border for the responsive dimensions between the desktop and iphone breakpoints - see below:

image

Shall we add a margin around the text and the video for the desktop breakpoint so that the text doesn't get pushed all the way to the edges?

**Comment by tynandebold** [comment]
Welcome @studioswong!

Yeah, great idea about the margin around the elements in these narrower widths. I agree it's a little squished. Do you mind if we hold off for now and sync with design about this tablet-ish design more generally? I think it could be good to get more guidance before potentially making a change now and then them saying something different, and it'll help move the other PRs along if we get this one in soonish.

**Review by tynandebold — APPROVED** [review]

File: components/hero/hero.module.scss
Comment (studioswong) [reply components/hero/hero.module.scss:62]: Thanks for sharing the github page - while I understand that css-modules might have recommended camel case, it is not enforced. 

I hope you don't mind my honesty, but I do find it quite odd first of all as this new naming convention isn't particularly brought up, and it has been our team norm to use BEM convention for css naming ( with many good reasons to, given that it is a [recommended best practice](https://cssguidelin.es/#naming-conventions) to follow for style naming). 

In looking at how the variables are set up in `variables.scss` here, it seems like we are adopting BEM naming here, so I am quite confused. 

While we might be using css-modules as a means to import styles in next.js, ultimately we are using scss which generally advocates [hyphen-delimited naming for classes ](https://sass-guidelin.es/#naming-conventions) - it doesn't to make sense to deviate from it just because we are using css-modules as a mechanism for style import.

**Review by studioswong — APPROVED** [review]
Happy to approve this to help this move on to the next PR 😄 

That said, I don't think there is particular need to wait for design to resolve the issue with the text eating into the edges as it's just a simple matter of adding a margin around the text for the large breakpoint, which we can derive from the design we have already below:

image

I also find it odd with this sudden change to allow camelCase in the css naming here, and given how our css variables are set up here in this PR, it is better if we stick to at least a hyphen delimited naming convention for scss classes here ( no need for strict BEM). I trust that the css class above would be resolved before being merged in. 😄 

Thank you!

File: components/hero/hero.tsx
Comment (studioswong) [reply components/hero/hero.tsx:4]: I see, thanks for clarifying!

File: components/hero/hero.module.scss
Comment (tynandebold) [reply components/hero/hero.module.scss]: I understand. I've changed it to a single-named class now.

To just touch on some of your other points: for the team norms point of view, I think we should be empowered and excited to try new things in this project. We don't need to carry over every convention from Kedro-Viz.

As for naming, I'm only following the convention that [Next.js documents](https://nextjs.org/docs/basic-features/built-in-css-support#sass-variables). They detail a SCSS variable, written with a hyphen, and then exported and used in JSX with camelCase. The same pattern would be used if it was a class name.

**Pull request merged**
